### PR TITLE
Aftershock: Shield bionics deploy Riemman shields

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary
Mods "Aftershock: Shield bionics deploy Riemman shields"

#### Purpose of change
Use the new energy shield functionality across the mod bionics. Makes it simpler to track player power across the mod.

#### Describe the solution
Removed and migrated all bionics but 3:

- Hawking Corps Mark I Shield: 120 shield hp + 0.5hp/s 1kw power draw
- Hawking Corps Mark V Shield: 550 shield hp + 2hp/s 2kw power draw
- Kinnetic Annulment System: 10000 shield hp + 20hp/s 150kw power draw.

All in all its pretty hard to keep these going unless you actively use a power generation bionic.

Since they now use the Riemman shell armor piece, they now provide full protection to all physical damage types.

#### Testing

Lots of testing including possible conflicts with the backpack generator I think I caught all the buggy interactions already, Making them turn each other off would be nice but not possible atm.

### CODE REVIEW

File: data/mods/Aftershock/items/armor/energy_shield.json [hunk 1]
```diff
@@ -102,9 +101,11 @@
     "type": "TOOL_ARMOR",
     "name": { "str": "SHD-1k Riemann Shield Generator (active)", "str_pl": "SHD-1k Riemann Shield Generators (active)" },
     "copy-from": "afs_backpack_shieldgen",
-    "description": "A backpack mounted shield generator for infantry use, capable of fully protecting its user during prolonged bursts of kinetic and energy attacks.  Standard issue for UICA breaching teams.\n\nIts high energy consumption and weight make it unsuitable for long-term use.  Powered with energy cartridges.",
-    "power_draw": "100000 mW",
+    "description": "A powerful shielding system salvaged from a pre-Discontinuity robotic warframe, crudely adapted for compatibility with biological hosts.  Powerful enough to withstand prolonged assault from anti vehicle weaponry.",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/items/armor/energy_shield.json:104]:
```suggestion
    "description": "A powerful shielding system salvaged from a pre-Discontinuity robotic warframe, crudely adapted for compatibility with biological hosts.  Powerful enough to withstand prolonged assault from anti-vehicle weaponry.",
```

File: data/mods/Aftershock/items/cbms.json [hunk 2]
```diff
@@ -232,63 +232,33 @@
...
     "type": "BIONIC_ITEM",
-    "name": { "str": "Holtzman Personal Shield Generator CBM" },
-    "description": "A military grade kinetic attenuation system developed for use within UICA's special service branches.  Although much more effective than anything that could be conceived with current technology, it remains a pale imitation of the Hyperspace Age systems it was reverse engineered from.  A side effect modern engineering seems unable to address is the tendency for this unit to release electrical fields when hit.  Activating the field will reduce incoming ballistic and piercing damage by 20 points.",
-    "price": "1 kUSD",
-    "price_postapoc": "65 USD",
+    "name": { "str": "HawkingCorps Mark V Shield CBM" },
+    "description": "Powerful and covert, the Mark V Shield is a military grade shielding system developed for use within UICA's special service branches.  Although much more effective than anything that could be conceived with current technology, it remains a pale imitation of the Hyperspace Age systems it was reverse engineered from.  A.\n\nActivating the shield will create an energy barrier capable of withstanding a modest amount of melee and ranged damage.",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/items/cbms.json:249]:
```suggestion
    "description": "Powerful and covert, the Mark V Shield is a military-grade shielding system developed for use within UICA's special service branches.  Although much more effective than anything that could be conceived with current technology, it remains a pale imitation of the Hyperspace Age systems it was reverse-engineered from.  A.\n\nActivating the shield will create an energy barrier capable of withstanding a modest amount of melee and ranged damage.",
```

File: data/mods/Aftershock/player/bionics.json [hunk 3]
```diff
@@ -232,84 +232,95 @@
...
+    "deactivated_eocs": [ "EOC_bio_stop_shield" ]
   },
   {
-    "id": "bio_forcefield_ballistic_medium",
+    "id": "bio_shield_medium",
     "type": "bionic",
-    "name": { "str": "Holtzman Personal Shield Generator" },
-    "description": "Personal shielding against bullets and sharp-pointed weapons comes in a variety of types.  This version is known for drastically increasing an individual soldier's lifespan while often decreasing that of anyone - friend or foe - in close proximity.  Activating the field will reduce incoming ballistic and piercing damage by 20 points.",
+    "name": { "str": "HawkingCorps Mark V Shield" },
+    "description": "Powerful and covert, the Mark V Shield is a military grade shielding system developed for use within UICA's special service branches.  Although much more effective than anything that could be conceived with current technology, it remains a pale imitation of the Hyperspace Age systems it was reverse engineered from.  A.\n\nActivating the shield will create an energy barrier capable of withstanding a modest amount of melee and ranged damage.",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/player/bionics.json:268]:
```suggestion
    "description": "Powerful and covert, the Mark V Shield is a military-grade shielding system developed for use within UICA's special service branches.  Although much more effective than anything that could be conceived with current technology, it remains a pale imitation of the Hyperspace Age systems it was reverse-engineered from.  A.\n\nActivating the shield will create an energy barrier capable of withstanding a modest amount of melee and ranged damage.",
```

File: data/mods/Aftershock/player/bionics.json [hunk 4]
```diff
@@ -232,84 +232,95 @@
...
+    "processed_eocs": [ "EOC_bio_update_ui" ],
+    "deactivated_eocs": [ "EOC_bio_stop_shield" ]
   },
   {
-    "id": "bio_forcefield_ballistic_heavy",
+    "id": "bio_shield_heavy",
     "type": "bionic",
     "name": { "str": "Kinetic Annulment System" },
-    "description": "A kinetic attenuation system salvaged from a pre-Discontinuity robotic warframe, crudely adapted for compatibility with biological hosts.  Powerful enough to completely stop most rifle cartridges.  Activating the field will reduce incoming ballistic and piercing damage by 40 points.",
+    "description": "A powerful shielding system salvaged from a pre-Discontinuity robotic warframe, crudely adapted for compatibility with biological hosts.  Powerful enough to withstand prolonged assault from anti vehicle weaponry.",
```
Comment (Night-Pryanik) [inline data/mods/Aftershock/player/bionics.json:299]:
```suggestion
    "description": "A powerful shielding system salvaged from a pre-Discontinuity robotic warframe, crudely adapted for compatibility with biological hosts.  Powerful enough to withstand prolonged assault from anti-vehicle weaponry.",
```

**Comment by John-Candlebury** [comment]
Imo its probably best to use the energy shield type armors over there too, since it offers extra levers for balancing in the long run. You could also make versions of them that only defend against specific damage types, I tested that works correctly in the first pr

I made everything defend against all damages here because the cognitive load is smaller for the player, since they have to track power and shield hp now.

**Pull request merged**